src/Engine/Utils.java

```diff
@@ -267,7 +269,7 @@ public static GamePath findShortestPath(XYCoord start, Unit unit, int x, int y,
    * @param y Final Y coordinate
    * @param map The map to search over.
    */
-  public static GamePath findShortestPath(XYCoord start, FloodFillFunctor fff, int initialFillPower, int x, int y, GameMap map)
+  public static GamePath findShortestPath(XYCoord start, Unit unit, MoveType fff, int initialFillPower, int x, int y, GameMap map, boolean canTravelThroughEnemies)
```

Comment:
Here and a few others - why require both a `Unit` and a `MoveType` when every unit has a `MoveType` already?

Reply:
`Unit` can be null, as discussed above.